core/src/window_service.rs

```diff
@@ -137,17 +141,50 @@ impl WindowServiceMetrics {
     }
 }
 
+fn should_send_index_and_erasure_conflicts(shred_slot: Slot, root_bank: &Arc<Bank>) -> bool {
+    match root_bank
+        .feature_set
+        .activated_slot(&feature_set::index_erasure_conflict_duplicate_proofs::id())
+    {
+        None => false,
+        Some(feature_slot) => {
+            let epoch_schedule = root_bank.epoch_schedule();
+            let feature_epoch = epoch_schedule.get_epoch(feature_slot);
+            let shred_epoch = epoch_schedule.get_epoch(shred_slot);
+            // Has a 1 epoch delay, as we don't have enough information
+            // on the epoch boundary of the feature activation
+            feature_epoch < shred_epoch
+        }
+    }
+}
+
 fn run_check_duplicate(
     cluster_info: &ClusterInfo,
     blockstore: &Blockstore,
     shred_receiver: &Receiver<PossibleDuplicateShred>,
     duplicate_slots_sender: &DuplicateSlotSender,
+    bank_forks: &RwLock<BankForks>,
 ) -> Result<()> {
+    let mut root_bank = bank_forks.read().unwrap().root_bank().clone();
```

Comment:
The `clone` is redundant:
https://github.com/solana-labs/solana/blob/2971e84ec/runtime/src/bank_forks.rs#L166